typeid/typeid-go/bench_test.go

```diff
@@ -349,6 +371,150 @@ func BenchmarkEncodeDecode(b *testing.B) {
 	}
 }
 
+// Benchmark Base32 operations directly
+func BenchmarkBase32(b *testing.B) {
+	b.Run("encode", func(b *testing.B) {
+		uid := uuid.Must(uuid.NewV7())
+		var bytes [16]byte
+		copy(bytes[:], uid.Bytes())
+		b.ReportAllocs()
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			_ = base32.Encode(bytes)
+		}
+	})
+	b.Run("decode", func(b *testing.B) {
+		uid := uuid.Must(uuid.NewV7())
+		var bytes [16]byte
+		copy(bytes[:], uid.Bytes())
+		encoded := base32.Encode(bytes)
+		b.ReportAllocs()
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			_, _ = base32.Decode(encoded)
+		}
+	})
+}
+
+// Benchmark memory usage with different batch sizes
+func BenchmarkMemoryUsage(b *testing.B) {
+	benchSizes := []int{100, 1000, 10000}
+	for _, size := range benchSizes {
+		b.Run(fmt.Sprintf("size=%d", size), func(b *testing.B) {
+			b.ReportAllocs()
+			for i := 0; i < b.N; i++ {
+				ids := make([]typeid.AnyID, size)
+				for j := range ids {
+					ids[j] = typeid.Must(typeid.WithPrefix("prefix"))
+				}
+				runtime.KeepAlive(ids)
+			}
+		})
+	}
+}
+
+// Benchmark parallel ID generation
+func BenchmarkParallelGeneration(b *testing.B) {
+	benchCases := []struct {
+		name       string
+		goroutines int
+		batchSize  int
+	}{
+		{"small_batch_few_goroutines", 4, 100},
+		{"small_batch_many_goroutines", runtime.GOMAXPROCS(0) * 2, 100},
+		{"large_batch_few_goroutines", 4, 1000},
+		{"large_batch_many_goroutines", runtime.GOMAXPROCS(0) * 2, 1000},
+	}
+
+	for _, bc := range benchCases {
+		b.Run(bc.name, func(b *testing.B) {
+			b.ReportAllocs()
+			b.ResetTimer()
+
+			for i := 0; i < b.N; i++ {
+				var wg sync.WaitGroup
+				wg.Add(bc.goroutines)
+
+				for g := 0; g < bc.goroutines; g++ {
+					go func() {
+						defer wg.Done()
+						for j := 0; j < bc.batchSize; j++ {
+							_ = typeid.Must(typeid.WithPrefix("prefix"))
+						}
+					}()
+				}
+
+				wg.Wait()
+			}
+		})
+	}
+}
+
+// Benchmark validation
+func BenchmarkValidation(b *testing.B) {
+	validIDs := make([]string, 100)
+	invalidIDs := make([]string, 100)
+
+	for i := range validIDs {
+		validIDs[i] = typeid.Must(typeid.WithPrefix("prefix")).String()
+		// Create invalid IDs by corrupting valid ones
+		if i < len(invalidIDs) {
+			invalidID := []byte(validIDs[i])
+			// Replace a character in the suffix to make it invalid
+			invalidID[len(invalidID)-1] = 'z'
+			invalidIDs[i] = string(invalidID)
+		}
+	}
+
+	b.Run("valid", func(b *testing.B) {
+		b.ReportAllocs()
+		for i := 0; i < b.N; i++ {
+			idx := i % len(validIDs)
```

Comment:
I don't think this will be accurate. Each iteration in `b.N` is supposed to do the same thing every time.